### ISSUE TITLE
Job Status first shows 'ABORTED' when binding a second SG with 2 DAs, then passes on subsequent Sync

### ISSUE THREAD

**Issue by mrajora-1** [opensecuritycontroller/osc-core#550]
### Description
Job Status first shows 'ABORTED' when binding a second SG with 2 DAs, then passes on subsequent Sync
### Expected Behavior
Jobs should have passed on the first attempt
### Actual Behavior
Job Status first shows 'ABORTED' when binding a second SG with 2 DAs, then passes on subsequent Sync
### Steps to Reproduce
- Create VC (using sfc), MC, 2DAs, 2DSs, 2SGs
- SG-1 is binded with da-1, da-2 (using SFC bind API)
- Bind SG-2 also with da-1, da-2 also  using SFC bind API
- The second bind shows an ABORTED job status. 
- Subsequent sync makes the job pass
### Additional Information
Excerpt from logs: 
2017-10-13 07:49:16,874 [ERROR| isc-job-pool-7| o.o.c.b.j.Job]: Fatal error during job execution (Job [name=Syncing Security Group 'sg-2', state=RUNNING, status=PASSED])
java.util.ConcurrentModificationException: null
	at java.util.HashMap$HashIterator.nextNode(HashMap.java:1437)
	at java.util.HashMap$EntryIterator.next(HashMap.java:1471)
	at java.util.HashMap$EntryIterator.next(HashMap.java:1469)
	at com.google.common.collect.AbstractMapBasedMultimap$Itr.next(AbstractMapBasedMultimap.java:1145)
	at org.osc.core.broker.job.Graph.getPredecessors(Graph.java:183)
	at org.osc.core.broker.job.TaskNode.getPredecessors(TaskNode.java:283)
	at org.osc.core.broker.job.Job.checkAllPredecessorsCompletedSuccessfully(Job.java:340)
	at org.osc.core.broker.job.Job.executeGraph(Job.java:245)
	at org.osc.core.broker.job.Job.run(Job.java:171)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:748)
### Environment
- OSC Version/Build: Version: 1.0.0-SNAPSHOT (Build:112-g42e1905, 2017-10-11T18:46:02Z)
- Cloud Provider Info:
- Plugin Info:
- Other:

### Status
- [ ] Reproduced
- [ ] In Progress
- [ ] Validated     =>    Verified in Version/Build: 

![aborted-job-when-binding-2nd-sg-with-sfc](https://user-images.githubusercontent.com/28600997/31561307-4d877bd2-b00c-11e7-8f46-a70922c27f52.JPG)

[securityBroker-ABORTED.log](https://github.com/opensecuritycontroller/osc-core/files/1383681/securityBroker-ABORTED.log)

**Comment by arvindn05** [comment]
Looks like the stacktrace is showing issue in the job engine...hence high priority/severity

**Comment by arvindn05** [comment]
DE5349

**Comment by dmitryintel** [comment]
The exact behavior is hard to reproduce. However,  we have added two unit tests to JobEngineTest, and these produce exactly the same stack trace. 

It appears the issue is with the tasks that modify their own task graph. If a new edge is added while getPredecessors is iterating through all the old edges, the iterator throws the ConcurrentModificationException.

We have reproduced the exception by adding two unit tests, which do not fail consistently! But after our fix, they pass consistently.
